@PROTOBUF_SYNTAX_VERSION@
import "goby/common/protobuf/option_extensions.proto";
import "goby/common/protobuf/app_base_config.proto";

package goby.common.protobuf;

message LiaisonConfig
{
  optional AppBaseConfig base = 1 [(goby.field).description="params shared with all goby applications"];

  optional string http_address = 2 [default="localhost", (goby.field).description="address to bind on; same as Wt --http-address"];
  optional uint32 http_port = 3 [default=54321, (goby.field).description="address to bind on; same as Wt --http-port"];

  optional string docroot = 4 [default="/usr/share/goby/liaison", (goby.field).description="path to static objects (e.g. images) root; same as Wt --docroot"];
  
  optional string additional_wt_http_params = 5 [(goby.field).example="--accesslog=/tmp/access.log"];
  optional float update_freq = 6 [default=5];

  repeated string load_shared_library = 7 [(goby.field).description="Path to a shared library containing compiled Protobuf files. Preferred over load_proto_file."];
  repeated string load_proto_file = 8 [(goby.field).description="Path to a .proto file. Use load_shared_library when possible."];
  repeated string load_proto_dir = 9 [(goby.field).description="Directory containing .proto files to load. Use load_shared_library when possible."];
  
  optional bool start_paused = 10 [default = false];

  optional string upper_right_logo = 11 [(goby.field).description="Image for upper right, relative to docroot"];
  optional string upper_right_logo_link = 12 [(goby.field).description="Hyperlink for upper right image"];
    
  // MOOS - 1000 to 1099
  // DRC -  1100 
  extensions 1000 to max;
}
